src/common/tracer.h

```diff
@@ -144,8 +144,32 @@ struct Tracer {
   jspan add_span(std::string_view span_name, const jspan_context& parent_ctx) { return {}; }
   void init(std::string_view service_name) {}
 };
-  inline void encode(const jspan_context& span, bufferlist& bl, uint64_t f=0) {}
-  inline void decode(jspan_context& span_ctx, ceph::buffer::list::const_iterator& bl) {}
+
+inline void encode(const jspan_context& span_ctx, bufferlist& bl, uint64_t f = 0) {
+  ENCODE_START(1, 1, bl);
+  // jaeger is missing, set "is_valid" to false.
+  bool is_valid = false;
+  encode(is_valid, bl);
+  ENCODE_FINISH(bl);
+}
+inline void decode(jspan_context& span_ctx, bufferlist::const_iterator& bl) {
+  DECODE_START(1, bl);
+  bool is_valid;
```

Comment:
> if the span was sent by another daemon that does have jaeger enabled, it would encode additional data when `is_valid` is true. we'd need to consume that as well

@cbodley Wouldn't `DECODE_FINISH` consume anything that's not consumed automatically?  What I had in mind was something like
```
inline void decode(jspan_context& span_ctx, bufferlist::const_iterator& bl) {
  DECODE_START(1, bl);
  // jaeger is missing, don't decode.
  DECODE_FINISH(bl);
}
```
to take advantage of `struct_compat` and `struct_len` checks provided by `DECODE_START`.   We could throw
```
  span_ctx = jspan_context();
```
at the end for a good measure but I don't think dummy decoding boilerplate is necessary.

Reply:
> Wouldn't `DECODE_FINISH` consume anything that's not consumed automatically?

@idryomov if that's the case, then i like your version better

Reply:
This was my recollection anyway.  Looking at the code, I think these lines would do it:
https://github.com/ceph/ceph/blob/34fc77ad585ad35dd6c0bbeea53b6fcd6b741571/src/include/encoding.h#L1513-L1514

Reply:
Thinking about this a bit more, I think `DECODE_START(1, bl)` should become `DECODE_START(254, bl)` to make sure that the "jaeger is missing" branch doesn't need to be tweaked when forward-compatible changes are made.

Reply:
Thanks for the suggestions! I've updated it.